src/reference/asciidoc/reactive-streams.adoc

```diff
@@ -160,12 +160,134 @@ When the target protocol for integration provides a Reactive Streams solution, i
+
+    private final CustomReactiveSource customReactiveSource;
+
+    public CustomReactiveSourceMessageProducer(CustomReactiveSource customReactiveSource) {
+        Assert.notNull(customReactiveSource, "'customReactiveSource' must not be null");
+        this.customReactiveSource = customReactiveSource;
+    }
+
+    @Override
+    protected void doStart() {
+        Flux < Message << ? >> messageFlux =
+            this.myReactiveSource
+            .map(event - >
+                MessageBuilder
+                .withPayload(event.getBody())
+                .setHeader(MyReactiveHeaders.SOURCE_NAME, event.getSourceName())
+                .build());
+
+        subscribeToPublisher(messageFlux);
+    }
+
+}
+```
+
+Usage would look like:
+
+```java
+public class MainFlow {
+  @Autowired
+  private CustomReactiveSourceMessageSource customReactiveSourceMessageSource;
+  
+  @Bean
+  public IntegrationFlow buildFlow() {
+     return IntegrationFlows.from(customReactiveSourceMessageSource)
+        .channel(outputChannel)
+        .get();
+  }
+}
+```
+Or in a declarative way:
+
+```java
+public class MainFlow {  
+  @Bean
+  public IntegrationFlow buildFlow() {
+     return IntegrationFlows.from(new CustomReactiveSourceMessageSource(new CustomReactiveSource()))
+        .handle(outputChannel)
+        .get();
+  }
+}
+```
+A reactive outbound channel adapter implementation is about the initiation (or continuation) of a reactive stream to interaction with an external system according to the provided reactive API for the target protocol.
+An inbound payload could be a reactive type per se or as an event of the whole integration flow which is a part of the reactive stream on top.
+A returned reactive type can be subscribed immediately if we are in a one-way, fire-and-forget scenario, or it is propagated downstream (request-reply scenarios) for further integration flow or an explicit subscription in the target business logic, but still downstream preserving reactive streams semantics.
+
+An example for a reactive outbound channel adapter:
+```java
+public class CustomReactiveMessageHandler extends AbstractReactiveMessageHandler {
+
+    private final CustomEntityOperations customEntityOperations;
+
+    private Expression queryTypeExpression = new ValueExpression<>(Type.INSERT);
+
+    public CustomReactiveMessageHandler(CustomEntityOperations customEntityOperations, Expression queryTypeExpression) {
+        Assert.notNull(customEntityOperations, "'customEntityOperations' must not be null");
+        this.customEntityOperations = customEntityOperations;
+        if (queryTypeExpression != null) {
+            this.queryTypeExpression = queryTypeExpression;
+        }
+    }
+
+    @Override
+    protected Mono<Void> handleMessageInternal(Message<?> message) {
+        return Mono.fromSupplier(() -> this.queryTypeExpression.getValue(this.evaluationContext, message, Type.class))
```

Comment:
@artembilan I'm struggling with the second example - can you simplify this to a more basic supplier usage that will be practical and won't require `evaluationContext`?

Reply:
Just do `-> message.getHeaders().get("queryType", Type.class)`.
And no `queryTypeExpression` in the code at all!